codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ChannelDuplexHandler.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.http2;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+
+/**
+ * A {@link ChannelDuplexHandler} providing additional functionality for HTTP/2. Specifically it allows to:
+ * <ul>
+ *     <li>Create new outbound streams using {@link #newStream()}.</li>
+ *     <li>Iterate over all active streams using {@link #forEachActiveStream(Http2Stream2Visitor)}.</li>
+ * </ul>
+ *
+ * <p>The {@link Http2FrameCodec} is required to be part of the {@link ChannelPipeline} before this handler is added,
+ * or else an {@link IllegalStateException} will be thrown.
+ */
+public class Http2ChannelDuplexHandler extends ChannelDuplexHandler {
+
+    private Http2FrameCodec frameCodec;
```

Comment:
@buchgr I think this should be volatile as we may call `newStream()` from multiple threads. That said we may also  be able to just check that the calling thread is correct via: `ctx.executor().inEventLoop(thread)`;